iib/workers/tasks/build.py

```diff
@@ -634,3 +641,63 @@ def handle_rm_request(
     _finish_request_post_build(
         output_pull_spec, request_id, arches, from_index, overwrite_from_index
     )
+
+
+@app.task
+def handle_regenerate_bundle_request(from_bundle_image, organization, request_id):
+    """
+    Coordinate the work needed to regenerate the operator bundle image.
+
+    :param str from_bundle_image: the pull specification of the bundle image to be regenerated.
+    :param str organization: the name of the organization the bundle should be regenerated for.
+    :param int request_id: the ID of the IIB build request
+    :raises IIBError: if the regenerate bundle image build fails.
+    """
+    _cleanup()
+
+    set_request_state(request_id, 'in_progress', 'Resolving from_bundle_image')
+    from_bundle_image_resolved = _get_resolved_image(from_bundle_image)
+    arches = _get_image_arches(from_bundle_image_resolved)
+    if not arches:
+        raise IIBError(
+            f'No arches were found in the resolved from_bundle_image {from_bundle_image_resolved}'
+        )
+
+    arches_str = ', '.join(sorted(arches))
+    log.debug('Set to regenerate the bundle image for the following arches: %s', arches_str)
+
+    payload = {
+        'from_bundle_image_resolved': from_bundle_image_resolved,
+        'state': 'in_progress',
+        'state_reason': f'Regenerating the bundle image for the following arches: {arches_str}',
+    }
+    exc_msg = 'Failed setting the resolved from_bundle_image on the request'
```

Comment:
`'Failed setting the resolved from_bundle_image on the request'` => `'Failed setting the resolved "from_bundle_image" on the request'`